File: app/src/main/java/com/example/miniproject/adapters/SourceRecyclerViewAdapter.java
package com.example.miniproject.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.miniproject.model.SourceModel;
import com.example.miniproject.R;

import java.util.List;

public class SourceRecyclerViewAdapter extends RecyclerView.Adapter<SourceRecyclerViewAdapter.ViewHolder> {
    private List<SourceModel> mSources;
    private Context mContext;

    public SourceRecyclerViewAdapter(Context mContext) {
        this.mContext = mContext;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(mContext)
                .inflate(R.layout.source_recyclerview_item, viewGroup, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int i) {
        SourceModel model = mSources.get(i);
        holder.sourceName.setText(model.getName());
        holder.sourceDescription.setText(model.getDescription());
        holder.category.setText(model.getCategory());
        holder.language.setText(model.getLanguage());
        holder.country.setText(model.getCountry());
        holder.parentCard.setTag(model);
        holder.parentCard.setOnClickListener((View.OnClickListener) mContext);
    }

    @Override
    public int getItemCount() {
        return mSources == null ? 0 : mSources.size();
    }

    public void setSourcesList(List<SourceModel> sourceModels) {
        mSources = sourceModels;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private RelativeLayout parentCard;
        private TextView sourceName, sourceDescription,
                language, country, category;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            parentCard = itemView.findViewById(R.id.sources_cardview_parent);
            sourceName = itemView.findViewById(R.id.source_name);
            sourceDescription = itemView.findViewById(R.id.description);
            language = itemView.findViewById(R.id.language);
            country = itemView.findViewById(R.id.country);
            category = itemView.findViewById(R.id.category);
        }
    }
}

File: app/src/main/java/com/example/miniproject/adapters/SourceRecyclerViewAdapter.kt
package com.example.miniproject.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.RecyclerView
import com.example.miniproject.R
import com.example.miniproject.model.SourceModel

class SourceRecyclerViewAdapter(private val mContext: Context) :
    RecyclerView.Adapter<SourceRecyclerViewAdapter.ViewHolder>() {
    private var mSources: List<SourceModel>? = null
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val view = LayoutInflater.from(mContext)
            .inflate(R.layout.source_recyclerview_item, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, i: Int) {
        mSources?.get(i)?.let { model ->
            holder.sourceName.text = model.name
            holder.sourceDescription.text = model.description
            holder.category.text = model.category
            holder.language.text = model.language
            holder.country.text = model.country
            holder.parentCard.tag = model
            holder.parentCard.setOnClickListener(mContext as View.OnClickListener)
        }
    }

    override fun getItemCount(): Int {
        return mSources?.size?:0
    }

    fun setSourcesList(sourceModels: List<SourceModel>?) {
        mSources = sourceModels
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val parentCard: ConstraintLayout = itemView.findViewById(R.id.sources_cardview_parent)
        val sourceName: TextView = itemView.findViewById(R.id.source_name)
        val sourceDescription: TextView = itemView.findViewById(R.id.description)
        val language: TextView = itemView.findViewById(R.id.language)
        val country: TextView = itemView.findViewById(R.id.country)
        val category: TextView = itemView.findViewById(R.id.category)

    }
}